Doc/library/sqlite3.rst

```diff
@@ -47,84 +47,148 @@ This document includes four main sections:
+Verify that the table has been created by running a query against it.
+Execute the query by calling :meth:`~Cursor.execute` on ``cur``,
+store the result in a variable ``res``,
+and call :meth:`~Cursor.fetchone` on ``res`` to fetch the first
+(and only) row from the query::
 
-   # We can also close the connection if we are done with it.
-   # Just be sure any changes have been committed or they will be lost.
-   con.close()
+   >>> res = cur.execute("SELECT count(rowid) FROM Movie")
+   >>> res.fetchone()
+   (0,)
 
-The saved data is persistent: it can be reloaded in a subsequent session even
-after restarting the Python interpreter::
+The result is a one-item :class:`tuple`:
+one row, with one column.
+It correctly shows a row count of zero.
 
-   import sqlite3
-   con = sqlite3.connect('example.db')
-   cur = con.cursor()
+Now, add two rows of data supplied as SQL literals
+by executing an ``INSERT`` statement,
+once again by calling :meth:`~Cursor.execute` on ``cur``::
 
-At this point, our database only contains one row::
+   cur.execute("""
+       INSERT INTO Movie VALUES
+           ('Monty Python and the Holy Grail', 1975, 8.2),
+           ('And Now for Something Completely Different', 1971, 7.5)
+   """)
 
-   >>> res = cur.execute('SELECT count(rowid) FROM stocks')
-   >>> print(res.fetchone())
-   (1,)
+The ``INSERT`` statement implicitly opens a transaction,
+which needs to be committed before changes are saved in the database
+(see :ref:`sqlite3-controlling-transactions` for details).
+Call :meth:`~Connection.commit` on the connection object
+to commit the transaction::
 
-The result is a one-item :class:`tuple`:
-one row, with one column.
-Now, let us insert three more rows of data,
-using :meth:`~Cursor.executemany`::
+   con.commit()
 
-   >>> data = [
-   ...    ('2006-03-28', 'BUY', 'IBM', 1000, 45.0),
-   ...    ('2006-04-05', 'BUY', 'MSFT', 1000, 72.0),
-   ...    ('2006-04-06', 'SELL', 'IBM', 500, 53.0),
-   ... ]
-   >>> cur.executemany('INSERT INTO stocks VALUES(?, ?, ?, ?, ?)', data)
+Execute a query to verify that the data was inserted correctly.
+Use the now familiar :meth:`~Cursor.execute` method on ``con``,
+store the result in ``res``,
+and call :meth:`~Cursor.fetchall` on ``res`` to fetch all rows::
+
+   >>> res = cur.execute("SELECT Score FROM Movie")
+   >>> res.fetchall()
+   [(8.2,), (7.5,)]
+
+The result is a :class:`list` with two rows of data,
+each a one-item :class:`tuple`,
+the single item representing the column queried.
+
+Now, insert three more rows by calling
+:meth:`~Cursor.executemany` on the cursor object,
+once again committing the transaction by calling :meth:`~Connection.commit`
+on the connection object::
+
+   data = [
+       ("Monty Python Live at the Hollywood Bowl", 1982, 7.9),
+       ("Monty Python's The Meaning of Life", 1983, 7.5),
+       ("Monty Python's Life of Brian", 1979, 8.0),
+   ]
+   cur.executemany("INSERT INTO Movie VALUES(?, ?, ?)", data)
+   con.commit()
```

Comment:
I would avoid repeating the part about `commit` in the prose, since it's somewhat distracting from the newly introduced `executemany`, and move it as a comment in the code where it appears in context after that the `executemany` is being dealt with.
```suggestion
Now, insert three more rows by calling
:meth:`~Cursor.executemany` on the cursor object::

   data = [
       ("Monty Python Live at the Hollywood Bowl", 1982, 7.9),
       ("Monty Python's The Meaning of Life", 1983, 7.5),
       ("Monty Python's Life of Brian", 1979, 8.0),
   ]
   cur.executemany("INSERT INTO Movie VALUES(?, ?, ?)", data)
   con.commit()  # remember to commit after executing the query
```